src/Container.js

```diff
@@ -0,0 +1,447 @@
+      console.warn(
+        'SpringRoll Container.openPath was passed a invalid options parameter. Using default parameters instead'
+      );
+      options = {};
+    }
+
+    this._internalOpen(
+      path,
+      Object.assign(
+        {
+          singlePlay: false,
+          playOptions: playOptions
+        },
+        options
+      )
+    );
+  }
+
+  /**
+   * Open application based on an API Call to SpringRoll Connect
+   * @param {string} api
+   * @param {object} options
+   * @param {string} [options.query='']
+   * @param {boolean} [options.singlePlay=false]
+   * @param {null | object} [playOptions=null]
+   * @memberof RemotePlugin
+   */
+  openRemote(api, { query = '', singlePlay = false } = {}, playOptions = null) {
+    this.release = null;
+
+    fetch(api).then(response => {
+      if (200 !== response.status) {
+        return;
+      }
+      response.json().then(release => {
+        const error = Features.test(release);
+        if (error) {
+          return this.client.trigger('unsupported');
+        }
+
+        this.release = release;
+        this._internalOpen(release.url + query, { singlePlay, playOptions });
+      });
+    });
+  }
+
+  /**
+   * Destroy and don't use after this
+   * @memberof Container
+   */
+  destroy() {
+    this.reset();
+    // Destroy in the reverse priority order
+    this.plugins
+      .slice()
+      .reverse()
+      .forEach(plugin => plugin.teardown(this));
+
+    this.main = null;
+    this.options = null;
+    this.dom = null;
+    this.release = null;
+  }
+
+  /**
+   * Tell the application to start closing
+   * @memberof Container
+   */
+  close() {
+    if (this.loading || this.loaded) {
+      this.plugins.forEach(plugin => plugin.close(this));
+      this.client.trigger('close');
+      // Start the close
+      this.client.send('close');
+    } else {
+      this.reset();
+    }
+  }
+
+  /**
+   * The current version of SpringRollContainer
+   * @readonly
+   * @static
+   * @memberof Container
+   */
+  static get version() {
+    return 'CONTAINER_VERSION';
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {object|function} plugin your plugin. This will be merged with a base plugin to make sure your plugin has certain functions
+   * @memberof Container
+   */
+  static uses(plugin) {
+    //Merging with the base plugin to guarantee we will have certain functions
+    const isConstructor = () =>
+      !!plugin.prototype &&
+      plugin.prototype.constructor &&
+      /^class/.test(plugin.prototype.constructor);
+
+    if ('function' !== typeof plugin && !isConstructor()) {
+      return;
+    }
+
+    PLUGINS.push(plugin);
+    PLUGINS.sort((a, b) => b.priority - a.priority);
+  }
+
+  /**
+   *
+   *
+   * @readonly
+   * @static
+   * @memberof Container
+   */
+  static get plugins() {
+    return PLUGINS;
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @memberof Container
+   */
+  static clearPlugins() {
+    PLUGINS.length = 0;
```

Comment:
setting length used to be faster, but I just ran a benchmark on jsperf and now a new empty array is the faster option?............¯\_(ツ)_/¯

Reply:
`PLUGINS = []` constructs a new array. In rare cases if it is referenced anywhere else in the code it can cause detached variable issues where in one place it still references the old array even though it should be empty. `PLUGINS.length = 0` is the recommended way as it will not create a new array and will continue to remove all of the contents.